### DISCUSSION TITLE
Change in behaviour upgrading cpr, with curl going from 7.80.0 to 8.10.0

### DISCUSSION THREAD

**Discussion by honeysuckle1234** [curl/curl#15386] [Q&A]
I have recently upgraded cpp project that's using cpr to perform GetAsync and PostAsync calls to a different service. Since this change, there has been a significant change in how these calls behave, they now seem to arrive in 1 minute batches or just aren't being sent immediately anymore. 

Is this there a possible explanation of this change in behaviour?

It is worth noting that nothing else other than the version of cpr has changed in this service, and the service it is making requests too has not changed at all.

**Comment by dfandrich** [comment]
Version 7.8 was released 23 years ago; are you sure that's the working version you
were using? Even if you meant 8.7.0, there have been 954 bug fixes and other
changes made between that version and 8.10.0. We try to maintain compatibility,
but sometimes we change undefined behaviour that applications are relying on.

I'd try upgrading to the latest release, 8.10.1, and see if the problem still
occurs before spending too much time debugging. If it does, we'll need a lot
more information about how the application is using curl to be able to help
much, including verbose debug logs and the code of the main transfer loop, as
well as an explanation of what cpr is.

**Reply by honeysuckle1234** [reply]
Sorry, should be 7.80.0

**Comment by bagder** [comment]
It would be most helpful if you could help narrow this down to an as small use case as possible that shows a difference and then you describe that in detail (if 8.10.1 still shows the problem). Ideally we can then convert that into a code without using cpr to rule out that it is something in the interaction between the components that cause the issue.

**Reply by bagder** [reply]
Also, if this is an *issue* then submit this as an issue, not a discussion.

**Comment by honeysuckle1234** [comment]
Not a curl issue, was a cpr issue.
